### ISSUE TITLE
Changing initial values resets form

### ISSUE THREAD

**Issue by sexualmaniac** [jaredpalmer/formik#188]
Hello, I have a form with initial values coming from redux. When values are changed there - my form resets itself to this values.

Is it a bug?

**Comment by dkrieger** [comment]
I'm having an issue that may be similar, but until/unless I can determine the root cause, I'll just piggy back here. My form resets every time my redux store changes, even though I'm not using my redux store to build the form.

Notes:

- I'm using withFormik HOC.
- I'm hardcoding my initial values directly in mapPropsToValues, rather than actually using props

Here's what I've determined:

- wrapping the withFormik-wrapped form with `pure` from recompose (performs shallow comparison in shouldComponentUpdate) seems to solve my problem.

Here's what I'm assuming as a result:

- it doesn't matter if props have changed, mapPropsToValues will be run any time componentWillReceiveProps would run.
- Formik assumes that your form will only receive (identical) props once. This assumption seems unreasonable. In a perfectly optimized react app, this assumption might be reasonable, but optimization is usually an after thought.

Here's what I'd recommend to remedy this issue:

- semantically, mapPropsToValues should only fire once, after mount -- changes to form values should only come from the form itself, not its parent.
- if there is a compelling reason to preserve existing functionality (can't really think of one), implement shouldComponentUpdate to shallow compare props and only return true if they are different.

If the maintainers have a better suggestion, or a justification for existing behavior, great! I found this behavior surprising.
